code/_lighting/SimpleLight.dm

```diff
@@ -209,33 +231,30 @@
 		destroy_medium_light()
 
 /atom/proc/update_medium_light_visibility()
-	if(src.medium_lights[1].invisibility == 101) // toggled off
-		return
-	if(!isturf(src.loc) && !isturf(src))
-		for (var/obj/overlay/simple_light/medium/light as anything in src.medium_lights)
-			src:vis_contents -= light
+	if(src.medium_lights[1].invisibility == INVIS_ALWAYS) // toggled off
 		return
-	for (var/obj/overlay/simple_light/medium/light as anything in src.medium_lights)
-		if(light.icon_state == "medium_center")
-			src:vis_contents += light
-			continue
-		var/turf/T = get_step(get_turf(src), light.dir)
-		if(T?.opacity || T?.opaque_atom_count)
+	if(isturf(src.loc) || isturf(src))
```

Comment:
You have now changed the logic. 
Was
        `if(!isturf(src.loc) && !isturf(src))`
Vs
`if(!(isturf(src.loc) || isturf(src)))`

Reply:
after checking, the logic works out because i swapped the blocks too